lib/fizzy/execute.cpp

```diff
@@ -152,15 +152,8 @@ inline uint32_t grow_memory(
     const auto new_pages = static_cast<uint32_t>(new_pages_u64);
 
     const uint64_t new_bytes = memory_pages_to_bytes(new_pages);
-    if constexpr (sizeof(size_t) < sizeof(uint64_t))
-    {
-#pragma clang diagnostic push
-#pragma clang diagnostic ignored "-Wtautological-type-limit-compare"
-        // NOLINTNEXTLINE(clang-diagnostic-tautological-type-limit-compare)
-        if (new_bytes > std::numeric_limits<size_t>::max())
-            return static_cast<uint32_t>(-1);
-#pragma clang diagnostic pop
-    }
+    if (!can_narrow_cast<size_t>(new_bytes))
```

Comment:
The GSL introduced `narrow_cast` and `narrow`. The `narrow` throws exception if conversion changes value. The `narrow_cast` is just an alias for `static_cast` so it always succeeds. So probably better name would be `can_narrow` or `can_safely_narrow`.